pkg/scheduler/podtask/port_mapping.go

```diff
@@ -2,168 +2,148 @@ package podtask
 
 import (
 	"fmt"
+	"sort"
 
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
-	log "github.com/golang/glog"
-	mesos "github.com/mesos/mesos-go/mesosproto"
 )
 
-type HostPortMappingType string
-
-const (
-	// maps a Container.HostPort to the same exact offered host port, ignores .HostPort = 0
-	HostPortMappingFixed HostPortMappingType = "fixed"
-	// same as HostPortMappingFixed, except that .HostPort of 0 are mapped to any port offered
-	HostPortMappingWildcard = "wildcard"
-)
+const PortMappingLabelKey = "k8s.mesosphere.io/portMapping"
 
-type HostPortMapper interface {
-	// abstracts the way that host ports are mapped to pod container ports
-	Generate(t *T, offer *mesos.Offer) ([]HostPortMapping, error)
-}
+// PortMapper represents a specifc port mapping strategy.
+type PortMapper string
 
-type HostPortMapping struct {
-	ContainerIdx int // index of the container in the pod spec
-	PortIdx      int // index of the port in a container's port spec
-	OfferPort    uint64
+// NewPortMapper returns an appropriate PortMapper for a given Pod.
+func NewPortMapper(pod *api.Pod) PortMapper {
+	filter := map[string]string{PortMappingLabelKey: "fixed"}
+	selector := labels.Set(filter).AsSelector()
+	if selector.Matches(labels.Set(pod.Labels)) {
+		return "fixed"
+	}
+	return "wildcard"
 }
 
-func (self HostPortMappingType) Generate(t *T, offer *mesos.Offer) ([]HostPortMapping, error) {
-	switch self {
-	case HostPortMappingWildcard:
-		return wildcardHostPortMapping(t, offer)
-	case HostPortMappingFixed:
+// PortMap computes PortMappings for the given Containers with the given offered port Ranges.
+func (m PortMapper) PortMap(offered Ranges, pod string, cs ...api.Container) ([]PortMapping, error) {
+	switch m {
+	case "wildcard":
+		return WildcardPorts(offered, pod, cs...)
+	case "fixed":
+		fallthrough
 	default:
-		log.Warningf("illegal host-port mapping spec %q, defaulting to %q", self, HostPortMappingFixed)
+		return FixedPorts(offered, pod, cs...)
 	}
-	return defaultHostPortMapping(t, offer)
 }
 
-type PortAllocationError struct {
-	PodId string
-	Ports []uint64
+// PortMapping represents a mapping from container ports to host ports for a
+// given pod container.
+type PortMapping struct {
+	PodName        string
```

Comment:
OK. I though that didn't matter too much here since there was a public `PodId` before.